components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/claim/InMemoryClaimManager.java

```diff
@@ -16,23 +16,126 @@
 
 package org.wso2.carbon.security.caas.user.core.claim;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.security.caas.user.core.bean.Attribute;
+import org.wso2.carbon.security.caas.user.core.bean.Domain;
 import org.wso2.carbon.security.caas.user.core.bean.User;
 import org.wso2.carbon.security.caas.user.core.exception.ClaimManagerException;
+import org.wso2.carbon.security.caas.user.core.exception.IdentityStoreException;
+import org.wso2.carbon.security.caas.user.core.exception.UserManagerException;
+import org.wso2.carbon.security.caas.user.core.store.connector.IdentityStoreConnector;
+import org.wso2.carbon.security.caas.user.core.user.UserManager;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 /**
  * InMemory Claim Manager.
  */
 public class InMemoryClaimManager implements ClaimManager {
 
+    private static final Logger log = LoggerFactory.getLogger(InMemoryClaimManager.class);
+
+    /**
+     * IdentityStoreConnector to MetaClaimMappings list.
+     */
+    private Map<String, List<MetaClaimMapping>> metaClaimMappingMap;
+
+    // TODO: Initialize userManager
+    private UserManager userManager;
+
+    // TODO: Initialize metaClaimStore
+    private MetaClaimStore metaClaimStore;
+
     @Override
     public List<Claim> getClaims(User user) throws ClaimManagerException {
-        return null;
+        List<Claim> claims = new ArrayList<>();
+        Domain domain = user.getDomain();
+
+        Map<String, List<MetaClaimMapping>> claimMappings = domain.getClaimMappings();
+
+        Map<String, IdentityStoreConnector> identityStoreConnectors = domain.getIdentityStoreConnectorMap();
+
+
+        for (Map.Entry<String, IdentityStoreConnector> identityStoreConnectorEntry :
```

Comment:
We cannot use streams here since I need to throw an Exception in the inner loop.